### ISSUE TITLE
Add option to have VS Code use OS scrollbars instead of custom scrollbars

### ISSUE THREAD

**Issue by SetTrend** [microsoft/vscode#130616]
It's currently cumbersome to scroll through alternating documents line by line, e.g. for comparison.

Adding scrollbar buttons (standard on Windows OS) to Visual Studio Code scrollbars would increase efficiency, speed, usability and accessibility.

So, I suggest to add scrollbar buttons to all VS Code scrollbars.

Either by ...
1. providing a Boolean option, switching between current custom scrollbars and OS scrollbars.
1. adding scrollbar buttons to the current custom scrollbars.

![Scrollbar buttons missing](https://user-images.githubusercontent.com/9283914/129106191-f383a998-0dbf-4e73-9dab-d83141dc76f7.gif)

**Comment by gjsjohnmurray** [comment]
Undocumented settings can do this for the editors:
```json
	"editor.scrollbar": {
		"verticalHasArrows": true,
		"horizontalHasArrows": true
	},
```

You need to reload the window after setting these.

**Comment by SetTrend** [comment]
Thanks for the hint!

However, wow, the current buttons are so titchy ... It takes too much time and attention (for me) to target and hit these with the mouse. They are not good to handle.

**Comment by isidorn** [comment]
@gjsjohnmurray thanks for jumping in. And yeah that is one option.

By default we are not rendering scroll bar buttons because it is not aligned with our design principles.
Though I do agree that in some cases they make it more usable.
However so far we have no plans to change this, since there is a workaround to use the keyboard or to use the editor hidden option. I can leave this open as a feature request so we gather more feedback.

**Comment by SetTrend** [comment]
Thanks, @isidorn, for taking the time.

Having more than two editors open, it's very cumbersome to change focus to the corresponding editor(group) first and then to scroll using the keyboard. In contrast to that, using the mouse it would be a simply click.

The current scrollbar button implementation is inacceptably hard to handle. Hitting a scrollbar button is a plain fluke:

![Scroll bar buttons tiny](https://user-images.githubusercontent.com/9283914/129204650-f077006f-8bac-4640-bacb-c1e3189b85f5.gif)

The most simple solution would be to have a Boolean option, disabling the custom scrollbars and using the OS provided scrollbars. That would remove all the current friction that's piggybacked with the current scrollbars and their odd (or hidden) behaviour.

**Comment by isidorn** [comment]
Thanks for feedback.
@hediet for future consideration if we decide to work on the scroll bar.

**Comment by SetTrend** [comment]
/ref: #123985 ["editor.scrollbar.scrollByPage" doesn't appear in settings](123985)

> I'm sorry about my slow response. As it is today, the `editor.scrollbar.scrollByPage` setting only applies to code editors and does not apply to all scrollbars in vscode. This setting (even when it was hidden) was only applying to the code editor. I think perhaps we could create global scrollbar settings i.e. `scrollbar.scrollByPage` at one point. Please feel free to create a new issue for adding support for this in all scrollbars.

_Originally posted by @alexdima in https://github.com/microsoft/vscode/issues/123985#issuecomment-901743489_

**Comment by SetTrend** [comment]
I tend to believe a title like _**"Add option to have VS Code use OS scrollbars instead of custom scrollbars"**_ would be more appropriate, as that option would be the most simple to implement solution to all anguish.

What would be your opinion on that change?

**Comment by isidorn** [comment]
I can rename the issue. However I am not sure about that exact change. I would like to keep this open to gather more user feedback.
The problem with OS scrollbars is that they would look out of style and not conforming to our design language. However if they provide better usability in some cases we should consider. Thus the reason to keep this open for more feedback.

**Comment by Zalfrin** [comment]
> Undocumented settings can do this for the editors:

Latest vscode release doesn't seem to support the mentioned settings.

+1 for this issue. editor.scrollbar.scrollByPage makes the scrollbar behave the way I want, but it needs to apply to the sidebar/etc scrollbars too.

**Comment by jasonzuvela-wywm** [comment]
yes please, stop dictating to users what they "want", and instead listen and see from the users' perspective what they actually want/need.

scrollbars without buttons (or with poorly implemented buttons, as shown in SetTrend's gif) are detracting from the user experience. lack of buttons reduces usability. (not to mention is inconsistent the rest of the whole operating system and application ecosystem)

if your "design principles" dictate not having buttons, then i would question those principles and the process used to determine them.

please add the buttons.

**Comment by Z-E-D** [comment]
I absolutely agree with this request. The missing arrow buttons on the scroll bars are the most annoying thing with VScode for me. After 30 years of using scroll bars with arrow buttons, it is really frustrating when you are forced to use another techniques to accomplish the same effect, i.e. to move the viewport just one line up/down at a time.

By the way, I disagree with the change of the title of this issue. Nobody wants to see OS scroll bars in VSCode, it would be out of place with its themes. You should just add two small arrow buttons using the same theme as the rest of the program.

I suppose the main problem here is because the scroll bars in the editors are overlaying the overview bar, which should be shrinked for the size of these newly added arrow buttons.

**Comment by Z-E-D** [comment]
[gjsjohnmurray](https://github.com/gjsjohnmurray):
> Undocumented settings can do this for the editors:
> 
> ```json
> 	"editor.scrollbar": {
> 		"verticalHasArrows": true,
> 		"horizontalHasArrows": true
> 	},
> ```
> 
> You need to reload the window after setting these.

As I mentioned in another issue, that modification didn't work for me as I expected, i.e. it made that the program doesn't respect my existing settings for the "editor.scrollbar.scrollByPage": true anymore. Well, it turned out that I shouldn't have copied the suggested lines directly as that, i.e. I should have merged my existing editor.scrollbar settings inside that group. Or, I could have add them just like this:
```
    "editor.scrollbar.verticalHasArrows": true,
    "editor.scrollbar.horizontalHasArrows": true,

```
And after I looked at the VSCode source code, I found one more undocumented setting that could be important to those who find those arrow buttons too small:
`    "editor.scrollbar.arrowSize": 20,
`

However, the problem with the current implementation is that the press on these buttons activates onMouseWheel event, i.e. it scrolls the editor view 2-3 lines instead of just one like the standard arrow buttons on scroll bars in every normal editor.

**Comment by SetTrend** [comment]
> "editor.scrollbar.arrowSize": 20,

Cool observation! 👍 

I'll add that to my VS Code settings template.

**Comment by VSCodeTriageBot** [comment]
We closed this issue because we [don't plan to address it](https://aka.ms/vscode-out-of-scope) in the foreseeable future. If you disagree and feel that this issue is crucial: we are happy to listen and to reconsider.

If you wonder what we are up to, please see our [roadmap](https://aka.ms/vscoderoadmap) and [issue reporting guidelines](https://aka.ms/vscodeissuereporting).

Thanks for your understanding, and happy coding!

**Comment by Krasen007** [comment]
I wonder if you guys will reconsider now that chatbots are reality and some time you want to scroll few pages of thinking to read what was happening?
I'm trying to find an extension or something like this to help me.

**Comment by wenma531** [comment]
> Adding scrollbar buttons (standard on Windows OS) to Visual Studio Code

Adding scrollbar buttons (standard on Windows OS) to Visual Studio Code

**Comment by wenma531** [comment]
> > Adding scrollbar buttons (standard on Windows OS) to Visual Studio Code
> 
> Adding scrollbar buttons (standard on Windows OS) to Visual Studio Code

I'm trying to find an extension
